### PR TITLE
Let macOS installer run without Rosetta on Apple Silicon

### LINKED ISSUE #16571: Fresh Installation of 7.2 (arm version) on Apple Silicon reqiures Rosetta
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Download and extract pwsh for MacOS (arm, currently powershell-7.2.0-osx-arm64.pkg) and try installation using the finder and right-click route.

### Expected behavior

```console
Rosetta-Less installation
```

### Actual behavior

```console
Installation requires Rosetta
```

### Error details

```console
To install "PowerShell - 7.2.0", you need to install Rosetta.
Do you want to install it now?

Rosetta enables Intel-based features to run on Apple silicon Macs. Reopening applications after installation is required to start using Rosetta.
```

### Environment data

```powershell
not applicable
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by dahlia**
# PR Summary

Let the macOS installer can run without Rosetta on Apple Silicon.

## PR Context

This patch makes the macOS installer (_powershell-\*-osx-arm64.pkg_) able to run without [Rosetta] on Apple Silicon (e.g., M1, M1 Pro, M1 Max).  Although the executable binary (*pwsh*) in itself already natively runs on Apple Silicon, the installer for it still requires Rosetta (as of 7.3.0-preview.1):

To install “PowerShell - 7.3.0-preview.1”, you need to install Rosetta.  Do you want to install it now?  Rosetta enables Intel-based features to run on Apple silicon Macs. Reopening applications after installation is required to start using Rosetta.

I fixed it by specifying the proper `hostArchitectures` to the `installer-gui-script` configuration.  (In PowerShell's packaging script, the template for it is placed in *tools/packaging/packaging.strings.psd1*.)  The solution was inspired by fish's fix commit: <https://github.com/fish-shell/fish-shell/commit/c1a1b7020356a4a8271542bd6f969670af260b42>.

Fixes <https://github.com/PowerShell/PowerShell/issues/16571>.

[Rosetta]: https://en.wikipedia.org/wiki/Rosetta_(software)#Rosetta_2

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -504,6 +504,29 @@ function Start-PSPackage {
...
+                    LTS = $LTS
+                    HostArchitecture = $HostArchitecture
+                }
+
+
+                if ($PSCmdlet.ShouldProcess("Create macOS Package")) {
+                    New-UnixPackage @Arguments
+                }
+            }
             default {
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:530]: is there any package type that doesn't have an explicit statement?  If not, we can replace default with a throw that this type is unknown

**Review by TravisEz13 — APPROVED** [review]
I don't see any blocking issues.

**Comment by iSazonov** [comment]
@dahlia Please rebase to fix CI-static.

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (dahlia) [reply tools/packaging/packaging.psm1:530]: I'm not sure if there are any other package types, as I have no much experience on this code base.  I'm going to let the default case throw an error that this type is unknown if you (or any other maintainers) ask!

**Comment by dahlia** [comment]
@iSazonov Rebased!

**Comment by dahlia** [comment]
Would there be more I need to adjust?

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (daxian-dbw) [reply tools/packaging/packaging.psm1:530]: @TravisEz13 anything you want @dahlia to change? If not, I will merge the PR as is.

Comment (iSazonov) [reply tools/packaging/packaging.psm1:530]: I guess it was a question for MSFT team about release process :-)

**Comment by daxian-dbw** [comment]
Chatted with @TravisEz13 offline and there is no blocker to merge this PR.
@dahlia Thanks for your contribution!

**Pull request merged**